utilities/Microsoft.Quantum.Katas/CheckKataMagic.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Jupyter.Core;
+using Microsoft.Quantum.IQSharp;
+using Microsoft.Quantum.IQSharp.Common;
+using Microsoft.Quantum.Simulation.Common;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Katas
+{
+    public class CheckKataMagic : MagicSymbol
+    {
+        /// <summary>
+        /// IQ# Magic that checks that the Reference implementation of a Kata runs successfully.
+        /// </summary>
+        public CheckKataMagic(IOperationResolver resolver, ICompilerService compiler, ILogger<KataMagic> logger)
+        {
+            this.Name = $"%check_kata";
+            this.Documentation = new Documentation() { Summary = "Checks the resference implementaiton of a single kata." };
+            this.Kind = SymbolKind.Magic;
+            this.Execute = this.Run;
+
+            this.Resolver = resolver;
+            this.Compiler = compiler;
+            this.Logger = logger;
+        }
+
+        /// <summary>
+        /// The Resolver let's us find compiled Q# operations from the workspace
+        /// </summary>
+        protected IOperationResolver Resolver { get; }
+
+        /// <summary>
+        /// The list of user-defined Q# code snippets from the notebook.
+        /// </summary>
+        protected ICompilerService Compiler { get; }
+
+        protected ILogger<KataMagic> Logger { get; }
+
+        /// <summary>
+        /// What this Magic does when triggered. It will:
+        /// - find the Kata to execute based on the Kata name,
```

Comment:
Does this work with katas (whole tutorials) or with tasks (individual tasks within a tutorial)? I think it's better to disambiguate right away

Reply:
@anpaz-msft Could you edit that as appropriate?